server/utils/security/command_safety.py

```
@@ -0,0 +1,251 @@
+            ensure_ascii=False,
+        )
+    )
+
+    try:
+        verdict = _call_llm(prompt, user_id, session_id)
+        if verdict.conclusion:
+            logger.warning(
+                "[CommandSafety] BLOCKED user=%s session=%s tool=%s cmd_fp=%s",
+                user_id, session_id, tool_name, _fingerprint(command),
+            )
+        return verdict
```

Ruff (0.15.11)

[warning] Consider moving this statement to an `else` block

(TRY300)